pkg/auth/auth.go

```diff
@@ -77,11 +79,12 @@ func Middleware(authenticator Authenticator, db ddb.Storage, idp IdentitySyncer)
 				return
 			}
 
-			// if we get ddb.ErrNoItems, the user may not have been synced yet from the IDP.
-			// try and sync them now.
-			// Note: this approach isn't very performant. In future this can be replaced with an incremental sync,
-			// which we can use for handling IDP webhook notifications rather than polling.
-			if err == ddb.ErrNoItems {
+			// // if we get ddb.ErrNoItems, the user may not have been synced yet from the IDP.
+			// // try and sync them now.
+			// // Note: this approach isn't very performant. In future this can be replaced with an incremental sync,
+			// // which we can use for handling IDP webhook notifications rather than polling.
+			if err == ddb.ErrNoItems && !isLocked {
+				isLocked = true
 				log.Info("user does not exist in database - running an IDP sync and trying again", "user", claims)
 				err = idp.Sync(ctx)
```

Comment:
@meyerjrr I'm wondering if this would be better implemented using a mutex in the idp.Sync(ctx) method